packages/hs-react-ui/src/components/Button/ButtonContainers.tsx

```diff
@@ -1,7 +1,96 @@
-import styled from 'styled-components';
-import { darken } from 'polished';
-import { ButtonTypes, ButtonContainer } from './Button';
+import styled, { StyledComponentBase, css, keyframes } from 'styled-components';
+import { darken, getContrast, complement } from 'polished';
 import colors from '../../enums/colors';
+import fonts from '../../enums/fonts';
+
+type StyledButtonProps = {
+  elevation: number;
+  color?: string;
+  fillColor?: string;
+};
+export enum ButtonTypes {
+  primary = 'primary',
+  secondary = 'secondary',
+  default = 'default',
+  destructive = 'destructive',
+  link = 'link',
+  outline = 'outline',
+}
+
+/**
+ * Returns color2 if color1 and color2 are contrasting, otherwise it returns the complementary color
+ * @param color1 The first color and color we want to have a >=4.5 contrast ration with
+ * @param color2 The second color and color we return if the contrast ration is >=4.5
+ */
+const getComplementaryColor = (color1: string, color2: string): string => {
+  const contrast = getContrast(color1, color2);
+
+  if (contrast > 4.5) {
+    return color2;
+  }
+
+  return complement(color1);
+};
+
+/* Keyframes for the loading bar gradient */
+const movingGradient = keyframes`
+  0% { background-position: 200% bottom; }
+  100% { background-position: 0% bottom; }
+`;
+
+/* Animation to scroll the gradient toward the right */
+const animation = css`
+  ${movingGradient} 8s linear infinite;
+`;
+
+/* Styled div that represents the scroll bar
+   Note: The border-radius 9999px is used to create a pill shape */
+export const Progress = styled.div`
+  ${() => css`
+    background: linear-gradient(
+        45deg,
+        rgba(255, 255, 255, 0.75),
+        rgba(0, 0, 0, 0.75),
+        rgba(255, 255, 255, 0.75),
+        rgba(0, 0, 0, 0.75),
+        rgba(255, 255, 255, 0.75)
+      )
+      repeat;
+    background-size: 400% 100%;
+    width: 5rem;
+    height: 10px;
+    margin-top: -5px;
+    margin-bottom: -5px;
+    border-radius: 9999px;
+    animation: ${animation};
+    line-height: 0;
+  `}
+`;
```

Comment:
Should this be centralized? Or would it be less code to maintain if we just use a <Text> which has the processing prop set to true?

Reply:
I think this should be centralized.

Let me explain what was going on here and maybe you have some thoughts on the situation. I was trying to use the `Text` component for loading and all of the other states as well, but what ended up happening is that the button was constantly adding padding around the child elements that are not plain text (even spans). I replicated the changes in Button.tsx adding slightly negative margins to keep the button the same size regardless of state. Using Text, the button would jump sizes to at least an extra 16px tall.